keel/Core/models.py

```diff
@@ -16,7 +16,7 @@ class Meta:
 
 
 class SoftDeleteModel(models.Model):
-    deleted_at = models.DateTimeField(blank=True, null=True)
+    deleted_at = models.DateTimeField(blank=True, null=True, db_index = True)
```

Comment:
do we filter on date values for deleted_at?

Reply:
to know if a row is soft-deleted or not... we always query on `deleted_at`. Thus index that field.

Reply:
so do we have a spcific date...or just check for null value
because indexes doesn't store null values

Reply:
??

Reply:
We only need to distinguish for null values.

Are you sure indexes will not help for null values queries?

Reply:
yes, totally ,...no RDBMS system stores null in indexes and neither the filtering helps

Reply:
Postgres does support this with newer versions, references below:

1. https://dba.stackexchange.com/questions/210292/does-postgresql-index-null-values

2. https://stackoverflow.com/questions/31966218/postgresql-create-an-index-to-quickly-distinguish-null-from-non-null-values

3. https://stackoverflow.com/questions/65621576/how-to-optimize-django-datetimefield-for-null-value-lookups-in-postgres

Reply:
We cant rely on specific versions for this.. anytime we need to switch  to another db engine or so

any waysnull lookups will still be inefficient if majority of data is null...
my point is we dont need a index for null values because optimiser itself is sufficient for null values lookup
 for time begin lets remove this index and migration supporting this